joomla/joomla-cms
# [4.0] Make sure the renderer does not manipulate the inline CSS and JS

Pull Request for Issue #28557

### Summary of Changes

Make sure there renderer does not manipulate the inline CSS and JS

### Testing Instructions

Add the following lines to the index.php of the cassiopea template:
```
// Add inline JavaScript
Factory::getDocument()->addScriptDeclaration('
    window.event("domready", function() {
        alert("An inline JavaScript Declaration");
    });
');

// Add inline Style
Factory::getDocument()->addStyleDeclaration('
	body {
		background: #00ff00;
		color: rgb(0,0,255);
	}
');
 ```

Enable CSP (System -> Content-Security-Policy -> Options) and configure like this:
![image](https://user-images.githubusercontent.com/2596554/79634500-0c88d800-816b-11ea-8746-24d09219d2d1.png)

- visit the frontend
- notice that the styles and scripts are blocked before the patch
- apply the patch
- notice it is working now

### Expected result

inline script and inline style tags are not modified by the renderer and can be whitelisted via an csp.

### Actual result

the style and script renderer add some spaces and line endings that breaks the CSP hash generation.

### Documentation Changes Required

none

cc @wilsonge @SharkyKZ
